app/src/controllers/reservation/services/ReservationService.ts

```diff
@@ -52,6 +57,47 @@ export const getNextAvailableDate = (reservationDate: Date, menuDuration: number
   return nextAvailableDate
 }
 
+export const checkInsideShopSchedule = (shopSchedule: {startTime?: string,
+  endTime?: string}, reservationDate: Date, menuDuration: number) => {
+  if (shopSchedule.startTime !== undefined
+    && shopSchedule.endTime !== undefined) {
+    const rEndTime = getNextAvailableDate(reservationDate, menuDuration)
+    const sStartTime = new Date(
+      `${reservationDate.getFullYear()}-${reservationDate.getMonth() + 1}-${reservationDate.getDate()}
+      ${shopSchedule.startTime}`,
+    )
+    const sEndTime = new Date(
+      `${reservationDate.getFullYear()}-${reservationDate.getMonth() + 1}-${reservationDate.getDate()}
+      ${shopSchedule.endTime}`,
+    )
+    if ((sStartTime.getTime() > reservationDate.getTime() || rEndTime.getTime() > sEndTime.getTime())
+      || (reservationDate.getTime() > sEndTime.getTime() - 3600000)
+    ) {
+      return false
+    }
+  }
+  return true
+}
+
+export const filterAndGetWantedDate = (reservationDate: Date) => {
```

Comment:
maybe change the name to `getStartAndEndDateFromReservationDate`? it's not really filtering anything, wanted date is too vague